### ISSUE TITLE
Add task generation script endpoint (EQ_generate_tasks)

### ISSUE THREAD

**Issue by mmcdermott** [payalchandak/EveryQuery#56]
Parent: #54

## Goal

Add a `EQ_generate_tasks` console script that generates task label parquets for EveryQuery pre-training. This is a step unique to EveryQuery (the AR model doesn't need separate task labels) — it pre-computes the `(subject_id, prediction_time, code, duration, occurs, censored)` tuples that the training loop samples from.

```bash
EQ_generate_tasks data_dir="$FINAL_DATA_DIR" \
    task_dir="$TASK_DIR" \
    codes_dir="$CODES_DIR" \
    +input_shard=0 +task_shard=0
```

## Current State

Two task generation implementations exist:

### 1. `sample_tasks.py` (sampling-first, newer)
- Samples N tasks from `codes × log-uniform(durations)`, then N×M patient contexts
- Parameterized by `(input_shard, task_shard, seed)` for Hydra `-m` sweeps
- Single `join_asof` evaluation per shard — efficient
- Has config: `sample_tasks_config.yaml`
- Entry: `@hydra.main(..., config_name="sample_tasks_config")`

### 2. `tasks.py` (exhaustive, older)
- Computes all valid prediction times per subject (≥50 prior events)
- Pre-computes time deltas to every query code
- Produces wide DataFrames per duration
- Has config: `tasks_config.yaml`
- Entry: `@hydra.main(..., config_name="tasks_config")`

Neither has a `[project.scripts]` entry point.

### pyproject.toml:
```toml
[project.scripts]
EQ_generate_tasks = "every_query.sample_tasks:main"
# Optionally also expose the exhaustive version:
EQ_generate_tasks_exhaustive = "every_query.tasks:main"
```

### Config path fix:
Both `sample_tasks.py` and `tasks.py` use `config_path="."` which resolves relative to the source file. This works in development but is fragile after pip install. Switch to:

```python
from importlib.resources import files
CONFIGS = files("every_query")

@hydra.main(version_base=None, config_path=str(CONFIGS), config_name="sample_tasks_config")
def main(cfg: DictConfig) -> None:
    ...
```

## Key Questions

**@gkondas** — Critical:
1. **Which task generator are you using for current experiments?** `sample_tasks.py` or `tasks.py`? Should external users use the sampling-first approach by default?
2. **How do you currently invoke task generation?** Via SLURM array jobs (`scripts/run_tasks.sh`)? Hydra multirun? What's the typical sweep shape (how many input shards × task shards)?
3. **Code lists** — task generation requires a set of query codes. How are these currently sourced? The `sample_codes/` directory has utilities for sampling code subsets. Should task generation include code sampling as a sub-step, or is that a separate prerequisite?
4. **Durations** — What durations are standard? The config shows `[30, 90, 180, 365, 731]` days in eval. Are these the same for training, or is the sampling-first approach's log-uniform sampling preferred?

**@payalchandak**:
1. The exhaustive `tasks.py` was the original approach. Is there a reason to keep both exposed, or should we consolidate to `sample_tasks.py` as the recommended path?
2. The `sample_codes/` utilities (`sample_train_codes.py`, `sample_eval_codes.py`) — are these prerequisites to task generation, or convenience utilities? Should they be part of this script or separate?

## Design Decisions

1. **One command vs. two**: Expose `sample_tasks.py` as the primary `EQ_generate_tasks`, and optionally keep `tasks.py` as `EQ_generate_tasks_exhaustive`? Or consolidate?
2. **SLURM integration**: The current `run_tasks.sh` scripts are SLURM-specific. External users may not have SLURM. The Hydra multirun approach (`-m input_shard=0,1,2,...`) is more portable.
3. **Code sampling**: Consider whether `EQ_generate_tasks` should accept either a code list file or a `sample_codes` sub-config to automatically select codes.

## Acceptance Criteria

- [ ] `EQ_generate_tasks` is a working console script after `pip install`
- [ ] Config path uses `importlib.resources.files` for package-relative resolution
- [ ] Works with Hydra multirun (`-m`) for parallel shard processing
- [ ] Documented in README with example invocation
- [ ] Tests pass (existing `test_sample_tasks.py` / `test_tasks.py` should still work)

**Comment by mmcdermott** [comment]
Resolved by #61 (merged in 34aa196). `EQ_generate_tasks` (= `every_query.sample_tasks:main`) and `EQ_generate_tasks_exhaustive` (= `every_query.tasks:main`) are both now installable console scripts.
